File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        Stack<Character> st = new Stack();
        int unb = 0;
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            if(c=='['){
                st.push(c);
            }
            else if(c==']'){
                if(st.empty()){
                    unb++;
                }
                else{
                    st.pop();
                }
            }
        }
        return (unb+1)/2;
    }
}

File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.py
class Solution:
    def minSwaps(self, s: str) -> int:
        stack=[]
        unb = 0
        for c in s:
            if c=='[':
                stack.append(c)
            else:
                if stack:
                    stack.pop()
                else:
                    unb+=1

        return (unb+1)//2